webapp/src/main/java/life/qbic/Application.java

```diff
@@ -93,21 +97,53 @@ private static MessageSubscriber whenUserRegisteredSendEmail(
     };
   }
 
+  private static MessageSubscriber whenPasswordResetRequestSendEmail(
+      ConfigurableApplicationContext appContext) {
+    return (message, messageParams) -> {
+      if (!messageParams.messageType.equals("PasswordReset")) {
+        return;
+      }
+      try {
+        var passwordResetRequest = deserializePasswordReset(message);
+        var passwordResetLink = appContext.getBean(EmailConfirmationLinkSupplier.class)
+            .emailConfirmationUrl(passwordResetRequest.userId().get());
+        var registrationEmailSender = appContext.getBean(
+            EmailService.class);
+        var passwordResetEmail = EmailFactory.registrationEmail("no-reply@qbic.life",
+            new Recipient(passwordResetRequest.userEmailAddress().get(),
+                passwordResetRequest.userFullName().get())
+            , passwordResetLink);
+        registrationEmailSender.send(passwordResetEmail);
+      } catch (IOException | ClassNotFoundException e) {
+        throw new RuntimeException(e);
+      }
+    };
+  }
+
   private static MessageSubscriber whenUserRegisteredLogUserInfo() {
     return (message, messageParams) -> {
       try {
-        UserRegistered userRegistered = deserialize(message);
+        UserRegistered userRegistered = deserializeUserRegistered(message);
         log.info(String.valueOf(userRegistered));
       } catch (IOException | ClassNotFoundException e) {
         log.error(e.getMessage(), e);
       }
     };
   }
 
-  static UserRegistered deserialize(String event) throws IOException, ClassNotFoundException {
+  static UserRegistered deserializeUserRegistered(String event)
+      throws IOException, ClassNotFoundException {
     byte[] content = Base64.decode(event);
     ByteArrayInputStream bais = new ByteArrayInputStream(content);
     ObjectInputStream ois = new ObjectInputStream(bais);
     return (UserRegistered) ois.readObject();
   }
+
+  static PasswordReset deserializePasswordReset(String event)
+      throws IOException, ClassNotFoundException {
+    byte[] content = Base64.decode(event);
+    ByteArrayInputStream bais = new ByteArrayInputStream(content);
+    ObjectInputStream ois = new ObjectInputStream(bais);
```

Comment:
but we dont want abstraction here, we want concrete types. I don't see how this can be improved tbh

Reply:
Hm, tbh I favor the explicit event type de-serialisation with a concrete return type instead of an open input stream passed around. That can easily lead to memory leak. And reading from an input stream is such a common thing, that there is no need to apply the DRY principle here. However the context exception handling is.